packages/authentication/cardstack/middleware.js

```diff
@@ -262,8 +262,13 @@ class Authentication {
   }
 
   async _applyReadAuthorization(session, user) {
+    let branch = this.controllingBranch.name;
     let schema = await this.currentSchema.forControllingBranch();
-    let authorizedUser = await schema.applyReadAuthorization(user, { session });
+    let documentContext = this.searcher.createDocumentContext({
+      branch,
+      schema,
+    });
+    let authorizedUser = await documentContext.applyReadAuthorization(user, { session });
```

Comment:
@balinterdi this doesnt make sense to me--why do you create a DocumentContext with no document? it seems as though the only reason to create a document context it to leverage the "ad hoc" read authorization against a document

Reply:
Yes, good point, I'm not sure. We discussed moving the the `_applyReadAuthorization` method here with @ef4 and at the time I understood some of the reasons. We don't have a document here that we could create a document context for, do we? Maybe `user` is one?

Reply:
I think `user` is the document in this case

Reply:
Yes, this code is applying read auth to the `user` document.